pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/test/java/org/apache/pinot/plugin/stream/pulsar/PulsarStandaloneCluster.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.stream.pulsar;
+
+import java.io.File;
+import org.apache.commons.io.FileUtils;
+
+
+public class PulsarStandaloneCluster {
+  public static final Integer DEFAULT_BROKER_PORT = 6650;
+  public static final Integer DEFAULT_ADMIN_PORT = 8080;
+  public static final String DEFAULT_DATA_MOUNT_DIRECTORY = "pulsar-data";
+  public static final String DEFAULT_CONF_MOUNT_DIRECTORY = "pulsar-conf";
+  public static final String DOCKER_CONTAINER_NAME = "pulsar_standalone_pinot";
+
+  private Integer _brokerPort;
+  private Integer _adminPort;
+  private String _dataMountDirectory;
+  private String _confMountDirectory;
+
+  private Process _pulsarCluster;
+
+  public static final String DOCKER_COMMAND =
```

Comment:
is there a way to do this without docker? something similar to how we bring up test Kafka cluster or test ZK?

Reply:
No, could not find an Admin API-based approach for this. Either docker or downloading the whole tar was the option.
The former seemed neat.

Reply:
@snleee @npawar Removed the Docker executor and now using standalone builder approach suggested by @snleee . It did lead to a lot of dependency convergence error and hence had to exclude so many dependencies and add their specific versions. 
I tried using pinot-parent versions for these dependencies but they are not working for almost all of them.

will try to prune as many dependencies as possible.

Reply:
done